src/controllers/campaign.js

```diff
@@ -513,26 +521,84 @@ const createMarketingCampaign = async (req, res) => {
 
             // add logic for other channels here
             if (campaignDataByChannel['Facebook']) {
-                for (const campaigns in campaignDataByChannel['Facebook']) {
-                    const _campaigns =
-                        campaignDataByChannel['Facebook'][campaigns];
-                    const totalCampaigns = _campaigns.length;
-                    for (let i = 0; i < totalCampaigns; i++) {
-                        const { name, id, type, budget, startDate, endDate } =
-                            _campaigns[i];
-                        await req.facebook.create(
-                            secret.FACEBOOK_ACCESS_TOKEN,
-                            facebookAdAccountId,
-                            {
-                                name,
-                                id,
-                                type,
-                                budget,
-                                startDate,
-                                endDate,
-                                status: 'PAUSED',
+                if (!facebookAdAccountId) {
+                    return res.status(400).json({
+                        message: `Invalid Facebook ADAccountId`,
+                    });
+                }
+                const { campaigns } = campaignAdSetAllocation.find(
+                    channel => channel.name === 'Facebook'
+                );
+
+                for (const campaign of campaigns) {
+                    try {
+                        const { name, id, campaignType, timePeriods } =
+                            campaign;
+                        const facebookCampaign =
+                            await req.facebook.createCampaign(
+                                secret.FACEBOOK_ACCESS_TOKEN,
+                                facebookAdAccountId,
+                                {
+                                    name,
+                                    id,
+                                    type: campaignType,
+                                    status: 'PAUSED',
+                                }
+                            );
+
+                        if (Array.isArray(timePeriods)) {
+                            for (const timePeriod of timePeriods) {
+                                for (const adset of timePeriod.adsets) {
+                                    try {
+                                        const {
+                                            name: adsetName,
+                                            bid_amount,
+                                            billing_event,
+                                            budget,
+                                            start_time,
+                                            end_time,
+                                            optimization_goal,
+                                            targeting,
+                                        } = adset;
+                                        await req.facebook.createAdset(
```

Comment:
Can you save the responses from the adset creation? so we can keep track of the adsets that were created or have failed

Reply:
Done!!